profiles/lib/tag_cache.lua

```diff
@@ -0,0 +1,35 @@
+TagCache = {}
+
+function TagCache.get(data, key)
+  local v = data.cache[key]
+  if v then
+    if v == '' then
```

Comment:
what's going on here is just that we always return nil in case the tag value is an empty string. this means we don't need to sprinkle the code with checks for empty strings.